docs/source/services/qos_service/enabling-rate-limiting.rst

```diff
@@ -0,0 +1,42 @@
+.. _enabling-rate-limiting:
+
+Enabling Rate Limiting
+======================
+
+Currently the QoS service allows a client to configure the read and write limits and enforces these limits.
+Detailed documentation is :ref:`here <qos-client-configuration>`.
+
+QoS also publishes client level metrics for the following:
+
+        - Meter readRequestCount
+        - Meter bytesRead
+        - Meter readTime
+        - Meter rowsRead
+        - Meter estimatedBytesRead
+        - Meter estimatedRowsRead
+        - Meter writeRequestCount
+        - Meter bytesWritten
+        - Meter writeTime
+        - Meter rowsWritten
+        - Meter backoffTime
+        - Meter rateLimitedExceptions
+
+A client should ideally look at the historical values of the ``bytesRead`` and ``bytesWritten`` metrics to determine the limits.
+
+How does QoS rate limit?
+------------------------
+
+We are using the standard Java Rate limiter to enforce the rate limiting with some modifications to support adjustment.
+It is not possible to estimate the number of bytes a read request is going to consume beforehand, so we request a
+default number of permits from the ratelimiter (currently 100) and subsequently return some permits (if the request
+returned less than a 100 bytes) or steal more permits (if the request returned more than a 100 bytes).
```

Comment:
steal -> consume

Reply:
Steal seems better her eas you are not consuming in the standard way to acquire permits. The rate-limiter has been modified to add new methods and this method is also named `steal`.

Reply:
Makes sense!